### PR TITLE
Support syntax for constant objects (strings, ints, floats)

### PR DESCRIPTION
**Pull request by fizruk**
Related to #554.

- [x] Support syntax for strings, ints, floats
- [x] Update built-in atoms to produce constant objects when possible
- [x] Do not desugar constant objects by default

Example:

```bash
cat test.phi
```
```
{⟦
  c ↦ Φ.org.eolang.float(as-bytes ↦ 25.0),
  result ↦
    ξ.c.times(x ↦ ⟦ Δ ⤍ 3F-FC-CC-CC-CC-CC-CC-CD ⟧)  // 1.8
       .plus(x ↦ ⟦ Δ ⤍ 40-40-00-00-00-00-00-00 ⟧), // 32.0
  λ ⤍ Package
⟧}
```

```bash
stack run -- dataize --recursive \
  -d eo-phi-normalizer/data/0.38.4/org/eolang/float.phi \
  -d eo-phi-normalizer/data/0.38.4/org/eolang/int.phi \
  -d eo-phi-normalizer/data/0.38.4/org/eolang/bytes.phi \
  -r eo-phi-normalizer/test/eo/phi/rules/yegor.yaml \
  test.phi
```
```
{
  ⟦
    c ↦ 25.0,
    result ↦ 77.0,
    λ ⤍ Package
  ⟧
}
```

### CODE REVIEW

File: eo-phi-normalizer/patent-draft.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Patent (draft)
```
Comment (deemp) [inline eo-phi-normalizer/patent-draft.md:1]: @fizruk, I don't think this file should be committed.

File: eo-phi-normalizer/src/Language/EO/Phi/Rules/Common.hs [hunk 2]
```diff
@@ -38,52 +38,17 @@ module Language.EO.Phi.Rules.Common where
...
-instance IsString Object where fromString = unsafeParseWith pObject
-instance IsString Binding where fromString = unsafeParseWith pBinding
-instance IsString Attribute where fromString = unsafeParseWith pAttribute
-instance IsString RuleAttribute where fromString = unsafeParseWith pRuleAttribute
-instance IsString PeeledObject where fromString = unsafeParseWith pPeeledObject
-instance IsString ObjectHead where fromString = unsafeParseWith pObjectHead
-
-instance IsString MetaId where fromString = unsafeParseWith pMetaId
-
-parseWith :: ([Token] -> Either String a) -> String -> Either String a
```
Comment (deemp) [inline eo-phi-normalizer/src/Language/EO/Phi/Rules/Common.hs:73]: @fizruk, here's the correct version of `parseWith`.

File: eo-phi-normalizer/src/Language/EO/Phi/Syntax.hs [hunk 3]
```diff
@@ -125,3 +227,242 @@ render d = rend 0 False (map ($ "") $ d []) ""
...
+instance IsString Object where fromString = unsafeParseWith pObject
+instance IsString Binding where fromString = unsafeParseWith pBinding
+instance IsString Attribute where fromString = unsafeParseWith pAttribute
+instance IsString RuleAttribute where fromString = unsafeParseWith pRuleAttribute
+instance IsString PeeledObject where fromString = unsafeParseWith pPeeledObject
+instance IsString ObjectHead where fromString = unsafeParseWith pObjectHead
+
+instance IsString MetaId where fromString = unsafeParseWith pMetaId
+
+parseWith :: ([Token] -> Either String a) -> String -> Either String a
```
Comment (deemp) [inline eo-phi-normalizer/src/Language/EO/Phi/Syntax.hs:457]: @fizruk, this is an incorrect version of `parseWith`.

**Review by deemp — CHANGES_REQUESTED** [review]
Pls address the comments.

**Comment by deemp** [comment]
Sugar isn't used sometimes.

```
eo-phi-normalizer rewrite --chain test.phi
```

```
Rule set following Nov 2024 revision
Input:
{
  ⟦
    c ↦ Φ.org.eolang.float (
      as-bytes ↦ 25.0
    ),
    result ↦ ξ.c.times (
      x ↦ ⟦
        Δ ⤍ 3F-FC-CC-CC-CC-CC-CC-CD
      ⟧
    )
    .plus (
      x ↦ ⟦
        Δ ⤍ 40-40-00-00-00-00-00-00
      ⟧
    ),
    λ ⤍ Package
  ⟧
}
====================================================
Result 1 out of 1:
[ 1 / 1 ] Normal form: {
  ⟦
    c ↦ Φ.org.eolang.float (
      as-bytes ↦ 25.0
    ),
    result ↦ ξ.c.times (
      x ↦ ⟦
        Δ ⤍ 3F-FC-CC-CC-CC-CC-CC-CD
      ⟧
    )
    .plus (
      x ↦ ⟦
        Δ ⤍ 40-40-00-00-00-00-00-00
      ⟧
    ),
    λ ⤍ Package
  ⟧
}
----------------------------------------------------
```

**Review by deemp — APPROVED** [review]

File: eo-phi-normalizer/src/Language/EO/Phi/Syntax.hs (hunk 3 above)
Comment (fizruk) [reply eo-phi-normalizer/src/Language/EO/Phi/Syntax.hs:457]: Wow, did not notice when resolving conflicts, sorry.

**Comment by 0crat** [comment]
@fizruk Hello team member, I noticed that the branch name "`constant-objects`" doesn't follow our [naming convention](https://www.yegor256.com/2014/04/15/github-guidelines.html). As per our policy, this results in a deduction of 12 points. In the future, please name your branches using the ticket number you're working on (e.g., "`601`"). This helps maintain consistency and clarity in our project management. Your current balance is now [-12](https://www.eolang.org/zerocracy/objectionary-vitals.html). Let's work together to improve our branch naming practices moving forward.

**Comment by 0crat** [comment]
@fizruk Thanks for the contribution! You've earned +4 points for this: +4 as a basis; -8 for too many hits-of-code (2766 >= 200); -16 for way too many hits-of-code (2766 >= 800); -4 for very few (3) comments; +28 to give you at least something. Please, [keep](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) them coming. Your running balance is [-24](https://www.eolang.org/zerocracy/objectionary-vitals.html).

**Comment by 0crat** [comment]
@deemp Great job on the review! 🎉 You've earned +10 points: +4 base, +16 for reviewing 2766 hits-of-code, with a slight -10 adjustment for only 4 comments. Keep up the good work! Your current balance is [+52](https://www.eolang.org/zerocracy/objectionary-vitals.html). Aim for more detailed feedback next time to maximize your points!

**Pull request merged**